packages/experiments/src/utilities/BaseComponentMin.ts

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { IStateComponentProps } from '../Foundation';
+import { IDisposable, IRefObject } from '../Utilities';
+
+/**
+ * BaseProps interface.
+ *
+ * @public
+ */
+// tslint:disable-next-line:no-any
+export interface IBaseProps<T = any> {
+  componentRef?: IRefObject<T>;
+}
+
+/**
+ * BaseComponent class, which provides basic helpers for all components.
+ *
+ * @public
+ */
+export class BaseComponentMin<TComponentProps extends IBaseProps = {}, TViewProps = {}, TState = {}> extends React.Component<
+  IStateComponentProps<TComponentProps, TViewProps>,
+  TState
+> {
+  /**
+   * Controls whether the componentRef prop will be resolved by this component instance. If you are
+   * implementing a passthrough (higher-order component), you would set this to false and pass through
+   * the props to the inner component, allowing it to resolve the componentRef.
+   */
+  protected _skipComponentRefResolution: boolean;
+
+  // tslint:disable:variable-name
+  private __disposables: IDisposable[] | null;
+  private __resolves: { [name: string]: (ref: React.ReactNode) => React.ReactNode };
```

Comment:
Do we need either of these for supporting componentRef?

Reply:
I think I don't need them if I trim the use cases down per your comment below.

Reply:
I think we can safely remove `__resolves`, but we need `__disposables` for `componentWillUnmount()`.